sycl/include/sycl/access/access.hpp

```diff
@@ -212,8 +212,9 @@ template <class T> struct remove_AS {
 
 #ifdef __SYCL_DEVICE_ONLY__
 template <class T> struct deduce_AS {
-  static_assert(!std::is_same<typename detail::remove_AS<T>::type, T>::value,
-                "Only types with address space attributes are supported");
+  // Undecorated pointers are considered generic.
```

Comment:
This is only legal when [generic as default address space](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#subsec:genericAddressSpace) mode is used. Should we add some kind of comment/`TODO`/`FIXME` to address [inferred address space](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#subsec:inferredAddressSpace) case (if we ever decide to go that way)?

Reply:
Good point. I have added a TODO.